interfaces/builtin/basedeclaration.go

```diff
@@ -345,6 +345,10 @@ slots:
       slot-snap-type:
         - core
     deny-auto-connection: true
+  maliit-input-method:
+    allow-installation:
+      slot-snap-type:
+        - app
```

Comment:
Please adjust this to be:
```
maliit-input-method:
    allow-installation:
      slot-snap-type:
        - app
    deny-connection: true
```

This makes it so that a snap declaration is required to connect to the slot implementation.

The above also allows plugging snaps to auto-connect which is something we only allow for slot implementations that ensure that connecting clients are properly isolated. Is this something that an unpatched maliit server ensures? Eg, does maliit protect against key sniffing, injection and the like? If not, please change the base declaration to:
```
maliit-input-method:
    allow-installation:
      slot-snap-type:
        - app
    deny-connection: true
    deny-auto-connection: true
```

Also if not, can you describe how the Canonical slot implementation will ensure this? If it is sufficiently hardened, then we can use this restrictive base declaration but issue a snap declaration that allows auto-connection to the Canonical snap.